newton/_src/viewer/recording.py

```
@@ -0,0 +1,375 @@
+            return videos_dir / "NewtonRecordings"
+        return Path.home() / "NewtonRecordings"
+
+    def suggested_output_path(self) -> Path:
+        """Return a timestamped default output path."""
+        stamp = datetime.now().strftime("%Y%m%d_%H%M%S")
+        return self.default_output_directory() / f"{self._filename_prefix}_{stamp}.mp4"
+
+    def _pick_encoder(self, ffmpeg_exe: str) -> str:
+        preferred = ("h264_nvenc", "h264_qsv", "h264_amf", "libx264")
+        try:
+            result = subprocess.run(
```

Ruff (0.15.13)

[error] `subprocess` call: check for execution of untrusted input

(S603)